version: "3.8"

services:
  json2video:
    build:
      context: .
    container_name: test-json2video
    image: test-json2video-image
    ports:
      - "5003:5003"
    volumes:
      - ./cache:/app/cache
      - ./movies:/app/movies
    environment:
      - FLASK_APP=app.run
      - PUBLIC_PORT=${MOVIES_PUBLIC_PORT}
      - CACHE_DIR=cache
      - MOVIES_DIR=movies
      - DEFAULT_ZOOM=0.002
      - SCHEME=https
      - CELERY_BROKER_URL=redis://test-redis:6379/0
      - CELERY_RESULT_BACKEND=redis://test-redis:6379/0
      - CELERY_LOG_LEVEL=ERROR
      - PYTHONPATH=/app
    networks:
      - app-network
    depends_on:
      - redis

  celery_worker:
    build:
      context: .
    container_name: test-celery-worker
    image: test-celery-worker-image
    command: celery -A app.celery_app worker --loglevel=error
    volumes:
      - ./cache:/app/cache
      - ./movies:/app/movies
    environment:
      - FLASK_APP=app.run
      - PUBLIC_PORT=${MOVIES_PUBLIC_PORT}
      - CACHE_DIR=cache
      - MOVIES_DIR=movies
      - DEFAULT_ZOOM=0.002
      - SCHEME=https
      - CELERY_BROKER_URL=redis://test-redis:6379/0
      - CELERY_RESULT_BACKEND=redis://test-redis:6379/0
      - CELERY_LOG_LEVEL=ERROR
      - PYTHONPATH=/app
    networks:
      - app-network
    depends_on:
      - redis

  redis:
    build:
      context: ./redis
      dockerfile: Dockerfile.redis
    container_name: test-redis
    image: test-redis-image
    ports:
      - "6379:6379"
    networks:
      - app-network
    privileged: true

  test:
    build:
      context: .
      dockerfile: Dockerfile.tests
    container_name: test-container
    image: test-image
    command: ["pytest", "-v"]
    volumes:
      - .:/app
      - /var/run/docker.sock:/var/run/docker.sock
    environment:
      - PYTHONPATH=/app
    networks:
      - app-network
    depends_on:
      - json2video
      - celery_worker
      - redis

networks:
  app-network:
    driver: bridge
